src/ServiceControl.Transports/TransportManifest.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace ServiceControl.Transports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+
+    public class TransportManifest
+    {
+        public string Version { get; set; }
+        public List<TransportManifestDefinition> Definitions { get; set; }
+    }
+
+    public class TransportManifestDefinition
+    {
+        public string Name { get; set; }
+        public string DisplayName { get; set; }
+        public string TypeName { get; set; }
+        public List<string> Aliases { get; set; }
+    }
+
+    public static class TransportManifestLibrary
+    {
+        public static List<TransportManifest> TransportManifests { get; set; }
+
+        static bool initialized;
+        static void Initialize()
+        {
+            if (TransportManifests == null)
+            {
+                TransportManifests = new List<TransportManifest>();
+            }
+
+            if (!initialized)
+            {
+                initialized = true;
+                var assemblyLocation = GetEntryOrExecutingAssemblyDirectory();
+                Directory.EnumerateFiles(assemblyLocation, "transport.manifest", SearchOption.AllDirectories).ToList().ForEach(manifest =>
+                {
+                    TransportManifests.Add(System.Text.Json.JsonSerializer.Deserialize<TransportManifest>(File.ReadAllText(manifest)));
+                });
+            }
+        }
+
+        static string GetEntryOrExecutingAssemblyDirectory()
+        {
+            var assemblyLocation = Assembly.GetEntryAssembly()?.Location ?? Assembly.GetExecutingAssembly().Location;
+            return Path.GetDirectoryName(assemblyLocation);
+        }
+
+        public static string Find(string transportType)
+        {
+            if (transportType == null)
+            {
+                throw new Exception("No transport has been configured. Either provide a Type or Name in the TransportType setting.");
+            }
+
+            Initialize();
+
+            var transportManifestDefinition = TransportManifests.SelectMany(t => t.Definitions).Where(w =>
+                    string.Compare(w.TypeName, transportType, true) == 0 || string.Compare(w.Name, transportType, true) == 0
```

Comment:
Might be worth lifting this test into `TransportManifestDefinition` as a method like `bool Match(string transportType)`. That way we can test that independently.

Reply:
changed